README.md

```diff
@@ -27,6 +29,17 @@ Metrics will now be accessible at http://localhost:8080/metrics
 
 To bind the java agent to a specific IP change the port number to `host:port`.
 
+__Note:__ Within the config.yaml...
+
+```
+hostPort: localhost:5555
+username: 
+password:
+```
+... is typically not required.
```

Comment:
I just created #675 which would make this config illegal in agent mode. So I guess we can remove this note now.

Reply:
I agree, https://github.com/prometheus/jmx_exporter/pull/675 makes it obsolete, but it _may_ be worth keeping for a new user since this seems like a common configuration error.

I have no strong opinion either way, but I know that @davidmpaz ran into this issue since the README wasn't clear.

Reply:
If we keep it we could replace it with a sentence saying _Note that the configuration file must not specify hostPort or jmxUrl when running in agent mode_.

Then we should also add the opposite note to the standalone section: _Note that the configuration file must specify hostPort or jmxUrl when running in standalone mode_.

The current _is typically not required_ is no longer true, because it will explicitly result in an error.